WIP: Asset class separate from Subscriber (#114)
ably/ably-asset-tracking-js: README.md

## Usage

### Subscribing to location updates

```ts
import { Subscriber } from '@ably/asset-tracking';

> // Client options passed to the underling ably-js instance

Reviewer: ```suggestion
// Client options passed to the underling ably-js instance.
```

Author: done

> // You must provide a clientId and some way for the client to authenticate with Ably

Reviewer: This is slightly misleading, considering token auth can include the client identifier in the token. Also, I think it could be a little more friendly. Perhaps...

```suggestion
// You must provide some way for the client to authenticate with Ably.
// In this example we're using basic authentication which means we must also provide a clientId.
// See: https://ably.com/docs/core-features/authentication
```

Author: very good point, fixed 😄

const ablyOptions = {
  key: ABLY_API_KEY,
  clientId: CLIENT_ID,
};

// Create a Subscriber instance
const subscriber = new Subscriber({
  ablyOptions,
});

// Get an asset
const asset = subscriber.get('my_tracking_id');

// Define a callback to be notified when a location update is recieved.
asset.addLocationListener((locationUpdate) => {
  console.log(`Location update recieved. Coordinates: ${locationUpdate.location.geometry.coordinates}`);
});

// Start tracking the asset. This will attach to the Ably realtime channel and enter presence.
await subscriber.start(trackingId);

> // Stop tracking the asset.

Reviewer: ```suggestion
// Stop tracking the asset, at some point later on when you no longer need to receive location updates.
```

Author: done

## Usage

### Subscribing to location updates

```ts
import { Subscriber } from '@ably/asset-tracking';

// Client options passed to the underling ably-js instance.
// You must provide some way for the client to authenticate with Ably.
// In this example we're using basic authentication which means we must also provide a clientId.
// See: https://ably.com/docs/core-features/authentication
const ablyOptions = {
  key: ABLY_API_KEY,
  clientId: CLIENT_ID,
};

// Create a Subscriber instance.
const subscriber = new Subscriber({
  ablyOptions,
});

// Get an asset.
const asset = subscriber.get('my_tracking_id');

// Define a callback to be notified when a location update is recieved.
asset.addLocationListener((locationUpdate) => {
  console.log(`Location update recieved. Coordinates: ${locationUpdate.location.geometry.coordinates}`);
});

// Start tracking the asset. This will attach to the Ably realtime channel and enter presence.

> await subscriber.start(trackingId);

Reviewer 2: Shouldn't it be `asset.start()`? 🤔 
```suggestion
await asset.start();
```

Author: yeah, you're right, i've fixed these now

// Stop tracking the asset, at some point later on when you no longer need to receive location updates.

> await subscriber.stop();

Reviewer 2: Like above, shouldn't it be an `asset`? 🤔 
```suggestion
await asset.stop();
```

```

### Subscribing to driver status

```ts
// Register a callback to be notified when the asset online status is updated.
asset.addStatusListener((isOnline) => {
  console.log(`Status update: Publisher is now ${isOnline ? 'online' : 'offline'}`);
});
```

### Requesting publisher resolution
```ts
import { Accuracy } from '@ably/asset-tracking';

// You can request a specific resolution to be considered by the publisher when you create an asset instance...
const resolution = {
  accuracy: Accuracy.High,
  desiredInterval: 1000,
  minimumDisplacement: 1,
};

const asset = subscriber.get('my_tracking_id', resolution);

// ...And you can send a request to change the resolution when the asset is already started

> await subscriber.sendChangeRequest({

Reviewer 2: Shouldn't it be an `asset` instead of a `subscriber`? 🤔 
```suggestion
await asset.sendChangeRequest({
```
